### PR TITLE
Fixed Sessions and Tweaked Code

### PR DESCRIPTION
**Pull request by javidangarcia**
- Fixed sessions so that when users log out it destroys the session.
- Added SearchQuery and SearchTable components for more code modularity.
- Tweaked some code based on feedback.

### CODE REVIEW

File: frontend/src/components/Search/Search.jsx [hunk 1]
```diff
@@ -80,63 +22,15 @@ export default function Search() {
...
-                                        </Link>
-                                    )
-                                }
-                            })
-                        }
-                    </div>
-                }
+                <SearchQuery searchInput={searchInput} />
             </div>
-            <table className="search-table">
```
Comment (vandanasub) [inline frontend/src/components/Search/Search.jsx:108]: nice, making these child components really improves this component's organization and readability

**Review by vandanasub — COMMENTED** [review]
Great better eng work and addressing comments :)

File: backend/routes/comments.js [hunk 2]
```diff
@@ -12,7 +12,8 @@ router.get("/comments/:ticker", async (req, res) => {
         const stock = await Stock.findOne({ where: { ticker } });
 
         if (stock === null) {
-            return res.status(404).json({ error: "This stock does not exist in the database." });
+            res.status(404).json({ error: "This stock does not exist in the database." });
+            return;
```
Comment (BriptimusPrimus) [inline backend/routes/comments.js:16]: 👍  Nice!

File: backend/routes/stocks.js [hunk 3]
```diff
@@ -67,24 +66,27 @@ router.get("/stocks/:ticker", async (req, res) => {
                 where: { UserId, StockId }
             });
 
-            stock.dataValues.following = following != null;
+            const currentStock = { ...stock.dataValues };
+
+            currentStock.following = following != null;
```
Comment (BriptimusPrimus) [inline backend/routes/stocks.js:71]: Well done. You have avoid mutating the stock object. 

Nit: You can even avoid mutations to the new object with:

```
const currentStock = { 
  ...stock.dataValues,
  following: following != null,
  liking: liking != null,
  disliking: disliking != null
};
```

File: frontend/src/components/App/App.jsx [hunk 4]
```diff
@@ -13,7 +13,7 @@ import Ranking from "../Ranking/Ranking";
 export const UserContext = createContext();
 
 export default function App() {
-    const [error, setError] = useState("");
+    const [errorMessage, setErrorMessage] = useState("");
```
Comment (BriptimusPrimus) [inline frontend/src/components/App/App.jsx:16]: 👍

File: frontend/src/components/Navbar/Navbar.jsx [hunk 5]
```diff
@@ -3,19 +3,27 @@ import axios from "axios";
...
-    const { user, setError } = useContext(UserContext);
+    const { user, setUser, setErrorMessage } = useContext(UserContext);
+    const navigate = useNavigate();
 
-    async function handleLogout() {
+    async function handleLogout(event) {
+        event.preventDefault();
         try {
-            await axios.post("http://localhost:3000/users/logout", {
+            const response = await axios.post("http://localhost:3000/users/logout", null, {
```
Comment (BriptimusPrimus) [inline frontend/src/components/Navbar/Navbar.jsx:15]: 👍  Very well done

File: frontend/src/components/Search/Search.jsx (hunk 1 above)
Comment (BriptimusPrimus) [reply frontend/src/components/Search/Search.jsx:108]: +1

**Comment by BriptimusPrimus** [comment]
Great job! These are massive improvements.

**Pull request merged**